get-all-systems-with-domains.py

```diff
@@ -0,0 +1,236 @@
+        help="Only process this amount of records",
+        type=int)
+    parser.add_argument(
+        "--split",
+        action="store_true",
+        default=False,
+        help="Split records between cloud.gov and Federalist domains"
+    )
+
+    return parser.parse_args()
+
+
+def get_all_instances(limit=0):
+    """
+    Retrieves all service instances that are associated with a domain configured
+    in the platform.
+    """
+
+    service_instances = []
+    resources_parsed = 0
+
+    page = CF_SERVICE_INSTANCES_API_URI
+
+    # If limit is set, display an extra message to remind the operator.
+    if limit > 0:
+        logger.info("Beginning API processing (limiting to {0} records)...".format(
+            limit
+        ))
+    else:
+        logger.info("Beginning API processing...")
+
+    while page is not None:
+        try:
+            output = json.loads(subprocess.check_output(["cf", "curl", page]).decode("utf-8"))
+        except subprocess.CalledProcessError as exc:
+            logger.error("Unable to execute cf curl: {0}".format(exc))
+            break
+
+        # Go through each resource returned and parse all of the relevant
+        # information we need.
+        for resource in output['resources']:
+            # Check to see if we've already hit the resource limit set when the
+            # script was invoked.
+            if limit > 0 and resources_parsed >= limit:
+                break
+
+            service_instances.append(parse_resource(resource))
+            resources_parsed = resources_parsed + 1
+
+        # Check to see if we've hit the limit of the amount of records we want
+        # to process.
+        if limit > 0 and resources_parsed < limit:
+            page = output["pagination"].get("next")
+        else:
+            page = None
+
+        # If we still have a page value, we need to retrieve the piece to get to
+        # the next page.
+        if page is not None:
+            uri_parts = page["href"].split("/")
+            page = "{0}/{1}".format(uri_parts[-2], uri_parts[-1])
+            logger.info("...Processing next page...")
+        else:
+            logger.info("...Finished processing.")
+
+    return service_instances
+
+
+def parse_resource(resource):
+    """
+    Takes in a CF API service instance resource and parses the following
+    information out of it:
+
+    - The broker name that manages the service instance
+    - The service plan name that was used with the broker
+    - The domain name associated with the service instance
+    - The created at date
+    - The space name that the service instance resides in
+    - The organization name that the service instance resides in
+    """
+
+    logger.info("    ...Parsing {0} ({1})...".format(
+        resource["name"],
+        resource["created_at"].split("T")[0]
+    ))
+
+    # Start pulling out the service instance information directly.
+    resource_info = {}
+    resource_info["domain_name"] = resource["name"]
```

Comment:
is this actually the domain name, or is this the service instance name?

Reply:
🤔 now that you point that out, technically speaking this is the service instance name name, but because of how the domain broker(s) work, the name seems to be a mirror of the domain (or the DNS record that is set for the domain).  I don't think the service instance metadata has the actual domain name in the output when I was looking and this was a close enough approximation.

Are you thinking it's better to be accurate in the description/key name of this, or somehow get the actual domain name?

Reply:
the name is whatever the user wants it to be - it just happens that our biggest user (federalist) is very, very consistent about naming

I think just renaming the variable so it's clear that it's _not_ the domain name is what we need (and the comment above)


This also reminded me: printing a header row would be good, too

Reply:
Sounds good!  And yes to the header row - I'll have it include it by default but include an option to omit just in case it's helpful.

Reply:
👨‍🍳❤️👌